In [1]:
#charger le fichier csv

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv(r"Student_Performance.csv")

In [7]:
df.head()

Hours Studied  Previous Scores Extracurricular Activities  Sleep Hours  \
0              7               99                        Yes            9   
1              4               82                         No            4   
2              8               51                        Yes            7   
3              5               52                        Yes            5   
4              7               75                         No            8   

   Sample Question Papers Practiced  Performance Index  
0                                 1               91.0  
1                                 2               65.0  
2                                 2               45.0  
3                                 2               36.0  
4                                 5               66.0

# Valeurs manqunates

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 6 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Hours Studied                     10000 non-null  int64  
 1   Previous Scores                   10000 non-null  int64  
 2   Extracurricular Activities        10000 non-null  object 
 3   Sleep Hours                       10000 non-null  int64  
 4   Sample Question Papers Practiced  10000 non-null  int64  
 5   Performance Index                 10000 non-null  float64
dtypes: float64(1), int64(4), object(1)
memory usage: 468.9+ KB


In [15]:
df.isna()

Hours Studied  Previous Scores  Extracurricular Activities  Sleep Hours  \
0             False            False                       False        False   
1             False            False                       False        False   
2             False            False                       False        False   
3             False            False                       False        False   
4             False            False                       False        False   
...             ...              ...                         ...          ...   
9995          False            False                       False        False   
9996          False            False                       False        False   
9997          False            False                       False        False   
9998          False            False                       False        False   
9999          False            False                       False        False   

      Sample Question Papers Practiced  Performance Index  
0                                False              False  
1                                False              False  
2                                False              False  
3                                False              False  
4                                False              False  
...                                ...                ...  
9995                             False              False  
9996                             False              False  
9997                             False              False  
9998                             False              False  
9999                             False              False  

[10000 rows x 6 columns]

In [16]:
df.isna().sum()

Hours Studied                       0
Previous Scores                     0
Extracurricular Activities          0
Sleep Hours                         0
Sample Question Papers Practiced    0
Performance Index                   0
dtype: int64

In [23]:
(df[df.duplicated()])

Hours Studied  Previous Scores Extracurricular Activities  Sleep Hours  \
915               9               52                         No            5   
1477              7               61                        Yes            6   
1601              5               99                         No            7   
1786              2               62                        Yes            9   
2026              5               87                        Yes            6   
...             ...              ...                        ...          ...   
9644              4               91                        Yes            4   
9940              8               95                         No            5   
9954              6               97                         No            8   
9966              1               41                         No            7   
9985              8               99                         No            5   

      Sample Question Papers Practiced  Performance Index  
915                                  9               48.0  
1477                                 8               54.0  
1601                                 5               89.0  
1786                                 4               40.0  
2026                                 7               74.0  
...                                ...                ...  
9644                                 3               71.0  
9940                                 2               90.0  
9954                                 7               92.0  
9966                                 3               12.0  
9985                                 5               92.0  

[127 rows x 6 columns]

In [56]:
condition = (df['Hours Studied'] == 9) & (df['Previous Scores'] == 52) & (df['Extracurricular Activities']=='No') & (df['Sleep Hours']==5)

In [57]:
df[condition]

Hours Studied  Previous Scores Extracurricular Activities  Sleep Hours  \
202              9               52                         No            5   
915              9               52                         No            5   

     Sample Question Papers Practiced  Performance Index  
202                                 9               48.0  
915                                 9               48.0

On remarque qu'il y a 127 lignes dupliquées , mais c'est pas un probleme car ilpeut y avoir dexu etudiants ayant ls memes caracteristiques .